dev/com.ibm.ws.security.wim.adapter.ldap/src/com/ibm/ws/security/wim/adapter/ldap/context/ContextManager.java

```diff
@@ -1603,6 +1769,97 @@ public String toString() {
         MISSING_PRIMARY_SERVER,
 
         /** Initialization succeeded with no errors. */
-        SUCCESS;
+        SUCCESS,
+
+        /**
+         * Initialization failed because a bindAuthMechanism was set to GSSAPI but either no password or an empty
+         * krb5PrincipalName was supplied.
+         */
+        MISSING_KRB5_PRINCIPAL_NAME;
+    }
+
+    /**
+     * Create a subject with the configure Kerberos information, to be used for
+     * creating a DirContext
+     *
+     * @return
+     */
+    private Subject handleKerberos() throws LoginException, MalformedURLException {
+        final String METHODNAME = "handleKerberos";
+        if (!isKerberosBindAuth()) {
+            if (tc.isDebugEnabled()) {
+                Tr.debug(tc, METHODNAME + " Kerberos login method was called, but Kerberos is not enabled. BindAuthMechanism is " + bindAuthMechanism);
+            }
+            return null;
+        }
+
+        if (kerberosService == null) {
+            if (tc.isDebugEnabled()) {
+                Tr.debug(tc, METHODNAME + " Kerberos login method was called, but the KerberosService is not enabled. BindAuthMechanism is " + bindAuthMechanism);
+            }
+            // todo kristip, throw exception?
```

Comment:
Seems if we don't have a kerberos service, then we should throw an exception. Perhaps this can be recreated with no krb5 configuration while bindAuthMechanism is set to GSSAPI? I don't believe that would stop us from starting up.

Reply:
I would change the above trace to be an error / warning as well and print out the same message as the exception contains.

Reply:
KerberosService has a default empty instance that starts, so I don't think we should hit a null because of a bad config, but clearly we'd be in a bad spot if we ended up here. I'll add an exception.